sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/utils/EdsProtocolTransformer.java

```diff
@@ -158,16 +150,24 @@ private static ServiceInstance transformEndpoint2Instance(LbEndpoint endpoint, S
         return instance;
     }
 
-    private static Optional<String> getServiceNameFromCluster(String clusterName) {
-        if (StringUtils.isEmpty(clusterName)) {
-            return Optional.empty();
+    private static String getServiceBaseClusterName(Map<String, XdsClusterLoadAssigment> instanceMap) {
+        for (Entry<String, XdsClusterLoadAssigment> instanceEntry : instanceMap.entrySet()) {
+            String[] splitCluster = instanceEntry.getKey().split(VERTICAL_LINE_SEPARATOR);
+            if (splitCluster[CLUSTER_SUBSET_INDEX].equals("")) {
```

Comment:
You are advised to check the size of the array.

Reply:
done